### PR TITLE
Default JDK and configurable test skippage

### PR DESCRIPTION
**Pull request by BertScholten**
- Switched from 11 to 17 as default JDK
- Allowing test failure behaviour to be configurable, and switched to default no ignoring.

### CODE REVIEW

File: events/release-event-action/action.yml [hunk 1]
```diff
@@ -25,7 +25,11 @@ inputs:
    JDK_VERSION:
      required: false
      description: "The version of the JDK to setup."
-     default: 11
+     default: 17
+  IGNORE_TEST_FAILURES:
+    required: false
+    description: "Allow tests to fail."
+    default: false
```
Comment (SerhatG) [inline events/release-event-action/action.yml]: I think this is the only one where we should force it not to ignore them, as it concerns a release.

Comment (BertScholten) [reply events/release-event-action/action.yml]: Myeah, probably a good idea. Can always disable the test if you get really desperate when releasing.

**Comment by SerhatG** [comment]
`JDK_VERSION` is also present in `extras/docker-build-or-publish-based-on-maven-project-version-action/action.yml` and `extras/docker-publish-action/action.yml`. Could you please update those as well.

Seems counter-intuitive, but this is the JDK version to setup for Maven. We use it to fetch the maven project version and do actions based on whether it's a SNAPSHOT/release version or not.

**Review by SerhatG — APPROVED** [review]
LGTM

**Pull request merged**